sources/iceCube/uhe/event/JulietEventGenerator.java

```diff
@@ -338,11 +418,21 @@ private void configureJULIeT(int flavorID, int doubletID, double energy, int med
         materialNumber = mediumID;
 
         // Generate Random Generator
-        rand = new RandomGenerator();
-        System.out.println("Random Generator has been generated");
+        if(seed != -1){
+          System.out.println("Using seed to setup Random Generator");
+          rand = new RandomGenerator(seed);
+        }
+        else if(random_state != null){
+          System.out.println("Using random_state to setup Random Generator");
+          rand = new RandomGenerator(random_state);
+        }
+        else{
+          System.out.println("Using system time to setup Random Generator");
+          rand = new RandomGenerator();
+        }
```

Comment:
I tried to move this part to the creation of the RandomGenerator. In this case if the seed -1 and no random_state is passed the old behavior is restored, which makes it a little easier to restore old behavior with less code from the juliet-interface side.

Reply:
I see what you did. Thanks. I think the changes you made seems to make sense.  Before merging into the master, why don't you introduce the similar modifications to JulietEventGeneratorGen2 class? Perhaps needs to some test run to ensure no new bugs are there.